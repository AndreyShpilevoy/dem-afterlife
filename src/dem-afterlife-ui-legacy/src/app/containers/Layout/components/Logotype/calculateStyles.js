const calculateStyles = ({logotype, focus}) => ({
    logotype: {
        backgroundImage: `url(${logotype.backgroundImage})`,
        backgroundRepeat: logotype.backgroundRepeat,
        backgroundSize: logotype.backgroundSize,
        height: '100%'
    },
    logotypeContainer: {
        height: logotype.height,
        '&:focus': {
            outlineColor: focus.colorLight
        }
    }
});

export default calculateStyles;