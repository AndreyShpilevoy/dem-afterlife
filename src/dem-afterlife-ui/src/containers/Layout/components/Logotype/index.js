import React from "react";
import { string, shape } from "prop-types";
import Link from "components/Link";
import { injectSheet } from "styles";
import calculateStyles from "./calculateStyles";

export const LogotypePure = ({ classes, className }) => {
  const linkClassName = `${classes.logotypeContainer} ${className || ""}`;
  return (
    <Link className={linkClassName} to="/">
      <div className={classes.logotype} />
    </Link>
  );
};

LogotypePure.displayName = "LogotypePure";

LogotypePure.propTypes = {
  classes: shape().isRequired,
  className: string
};

LogotypePure.defaultProps = {
  className: ""
};

export default injectSheet(calculateStyles, { componentName: "Logotype" })(
  LogotypePure
);
