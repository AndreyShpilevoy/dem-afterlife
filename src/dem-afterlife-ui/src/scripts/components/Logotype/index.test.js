/* eslint no-undef: 'off'*/
/* eslint import/no-extraneous-dependencies: 'off' */

import React from 'react';
import {shallow} from 'enzyme';
import Logotype, {LogotypePure} from './index';

describe('Logotype', () => {
    const hocProps = {
        theme: 'default',
        classNames:
        {
            logotype: 'logotype-0-0',
            logotypeContainer: 'logotypeContainer-0-1'
        }
    };

    it('HOC match expected snapshot', () => {
        expect(shallow(<Logotype theme={hocProps.theme} />)).toMatchSnapshot();
    });

    it('Pure component match expected snapshot', () => {
        expect(shallow(<LogotypePure classNames={hocProps.classNames}/>)).toMatchSnapshot();
    });
});
