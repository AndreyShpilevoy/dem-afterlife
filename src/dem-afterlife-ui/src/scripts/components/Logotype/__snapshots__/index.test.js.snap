// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logotype HOC match expected snapshot 1`] = `
<LogotypePure
  classNames={
    Object {
      "logotype": "logotype-0-0",
      "logotypeContainer": "logotypeContainer-0-1",
    }
  }
  theme="default"
/>
`;

exports[`Logotype Pure component match expected snapshot 1`] = `
<Link
  className="logotypeContainer-0-1"
  replace={false}
  to="/"
>
  <div
    className="logotype-0-0"
  />
</Link>
`;
